### PR TITLE
feat: add global prop defaults, update bootstrapping

### PR DESCRIPTION
**Pull request by nekosaur**

### CODE REVIEW

File: packages/vuetify/src/framework.ts [hunk 1]
```diff
@@ -1,83 +1,112 @@
...
+  }
+
+  return {
+    install,
   }
+}
+
+// Would be nice to have PropOptions here
+const isFactory = (type: any) => {
+  const arr = Array.isArray(type) ? type : [type]
```
Comment (johnleider) [inline packages/vuetify/src/framework.ts:75]: could we use the `wrapInArray` helper function?

File: packages/vuetify/src/framework.ts [hunk 2]
```diff
@@ -1,83 +1,112 @@
...
     }
 
-    // rtl is not installed and
-    // will never be called by
-    // the init process
-    this.framework.rtl = Boolean(this.preset.rtl) as any
+    app.provide(VuetifySymbol, vuetify)
+  }
+
+  return {
```
Comment (johnleider) [inline packages/vuetify/src/framework.ts:68]: Could be a single line.

File: packages/vuetify/src/framework.ts [hunk 3]
```diff
@@ -1,83 +1,112 @@
...
+const isFactory = (type: any) => {
+  const arr = Array.isArray(type) ? type : [type]
+  return arr.find((t: any) => t === Array || t === Object)
+}
 
-  mergePreset ({ preset, ...userPreset }: VuetifyUserPreset) {
-    return mergeDeep(mergeDeep(Preset, preset), userPreset) as VuetifyPreset
+export const makeProps = <P extends Record<string, Prop<any>>>(props: P) => {
+  for (const key in props) {
+    const propOptions = (props[key] as any)
```
Comment (johnleider) [inline packages/vuetify/src/framework.ts:81]:
```suggestion
    const propOptions = (props[key] as any)

```

File: packages/vuetify/src/framework.ts [hunk 4]
```diff
@@ -1,83 +1,112 @@
...
-  // Instantiate a VuetifyService
-  use (Service: VuetifyService) {
-    const property = Service.property
+  return props
+}
 
-    if (property in this.framework) return
+const generateDefault = (propName: string, localDefault: any, type: any) => {
+  return () => {
+    const vm = getCurrentInstance()
```
Comment (johnleider) [inline packages/vuetify/src/framework.ts:90]:
```suggestion
    const vm = getCurrentInstance()

```

File: packages/vuetify/src/framework.ts [hunk 5]
```diff
@@ -1,83 +1,112 @@
...
+    if (!vm || !vm.type.name) {
+      console.warn('Unable to get current component instance when generating default prop value')
```
Comment (johnleider) [inline packages/vuetify/src/framework.ts:92]:
```suggestion
      console.warn('Unable to get current component instance when generating default prop value')

```

File: packages/vuetify/src/framework.ts [hunk 6]
```diff
@@ -1,83 +1,112 @@
...
+      return localDefault
+    }
```
Comment (johnleider) [inline packages/vuetify/src/framework.ts:94]:
```suggestion
    }

```

File: packages/vuetify/src/framework.ts [hunk 7]
```diff
@@ -1,83 +1,112 @@
...
 }
+
+// TODO: Fix typings and shit
+const getGlobalDefault = <
+  C extends string,
+  P extends string
+>(component: C, prop: P) => {
+  const vuetify = useVuetify()
+  const key = component as keyof VuetifyComponentDefaults
+  const defaults = vuetify.defaults[key] as any
```
Comment (johnleider) [inline packages/vuetify/src/framework.ts:109]:
```suggestion
  const defaults = vuetify.defaults[key] as any

```

File: packages/vuetify/src/framework.ts [hunk 8]
```diff
@@ -1,83 +1,112 @@
...
+  if (defaults == null) return null
```
Comment (johnleider) [inline packages/vuetify/src/framework.ts:110]:
```suggestion
  if (defaults == null) return null

```

**Review by johnleider — CHANGES_REQUESTED** [review]

**Review by KaelWD — APPROVED** [review]

**Pull request merged**